### DISCUSSION TITLE
Markers CCTAG with Python.

### DISCUSSION THREAD

**Discussion by AdonaiVera** [stella-cv/stella_vslam#433] [Q&A] [answered]
Hi Stella Vslam Team, 
First, thank you for this fantastic project; I'm trying to use CCTAG to scale my point cloud. I know that there is already an implementation with AURco; However, I need to use CCTAG; the process that I'm doing in Python is the next one:
1. I use the .map file to get 3d points, rotation, and translation matrix from camera to world and vice-versa. 
2. I found the marker on the images, and I get the X and Y of the equirectangular image. 
3. I'm transforming from Pixel to bearing. 
4. Then I have to transform to coordinate worlds. (I'm trying to apply some Rigidly (+scale) to align two point clouds with known point-to-point correspondences with least-squares error. (I have the pixels of the 3d points that I get from opevslam, and I can create a new rotation and translation matrix using this relation)
5. I take the bearing and found the marker in the world coordinate.

My results are inaccurate; I'm trying to understand how you find the marker in the world coordinate system; any advice would be excellent.
My issue is in step four; I believe there is another way to go from bearing to the world coordinate. 

Thank you

**Comment by ymd-stella** [comment]
What is Rigidly?

**Reply by AdonaiVera** [reply]
Rigid transformation

**Comment by ymd-stella** [comment, marked as answer]
It is recommended to inherit `marker_detector::base` and add `marker_detector::cctag`.

(Not sure what you mean by "with Python.")

**Reply by AdonaiVera** [reply]
I'm creating all the modules in Python. I only use the msg file that I generated using Stella vslam. However, I'll execute it first in c++, as you suggest.
Thank you
